### PR TITLE
Adapt Espresso tests to new public link 

### PR DESCRIPTION
**Pull request by jesmrec**
New public link options: https://github.com/owncloud/android/pull/2150

Espresso tests fixed to adapt this new UI. Included fix for the welcome wizard and removed unused code.

### CODE REVIEW

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 1]
```diff
@@ -227,35 +231,39 @@ public void test_01_create_public_link_folder_defaults()
...
-        //Depending the server version, send a name or not.
-        if (capabilities.getVersionMayor() >= VERSION_10) {
-            publicShareCreationAllEnabled(nameShare);
-        } else {
-            publicShareCreationAllEnabled(null);
-        }
+        //Enable the option for Download/View/Upload permissions
+        onView(withId(R.id.shareViaLinkEditPermissionReadOnly)).perform(click());
 
-        //Check the name,only in the case of ownCloud >= 10
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: Neither this?

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 2]
```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
...
 
-        Log_OC.i(LOG_TAG, "Test Share Public All Enabled Passed");
+        Log_OC.i(LOG_TAG, "Test Share Public with Download/View");
 
     }
 
     /**
-     *  TEST CASE: Share public a folder and gets the link
-     *  PASSED IF: Link correctly copied in clipboard
+     *  TEST CASE: Share publicly a folder woth DownloadV/View/Upload permission
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: Typo in Download**V**/View/Upload permission

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 3]
```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
...
+     *  PASSED IF: Link created and visible in share view with Download/View/Upload option
+     *
      */
     @Test
-    public void test_03_get_link()
+    public void test_03_create_public_link_download_view_permission()
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: What about renaming the test to `test_03_create_public_link_download_view_upload_permission`?

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 4]
```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
...
 
-        //if ownCloud >= 10, we can handle the link name. If not... skipping test.
-        if (capabilities.getVersionMayor() >= VERSION_10) {
+        //The message of "not links created yet" is gone
+        onView(withId(R.id.shareNoPublicLinks))
+                .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.GONE)));
 
-            //Select share option
-            selectShare(folder);
+        onView(withId(R.id.editPublicLinkButton)).perform(click());
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: Should we edit the share in this test or is just legacy code from when the test name was `test_04_edit_name`?

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 5]
```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
...
-        Log_OC.i(LOG_TAG, "Test Edit Link Name Passed");
     }
 
+
     /**
-     *  TEST CASE: Edit the public folder by enabling "Allow Editing"
-     *  PASSED IF:
-     *             - "Allow editing" and "Show file listing" (oC >= 10) are enabled.
-     *             - "Password" and "Expiration date" are disabled.
+     *  TEST CASE: Share publicly a folder woth DownloadV/View/Upload permission
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: woth DownloadV/View/Upload => with Upload only permission

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 6]
```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
...
+        onView(withId(R.id.shareNoPublicLinks)).check(matches(isDisplayed()));
+        onView(withId(R.id.addPublicLinkButton)).perform(click());
+        onView(withId(R.id.shareViaLinkNameValue)).perform(replaceText(nameShare));
 
-        //Check file listing disabled + checked (default) if ownCloud >= 10
-        if (isSupportedFileListing()) {
-            onView(withId(R.id.shareViaShowFileListingSwitch)).check(matches(not(isEnabled())));
-            onView(withId(R.id.shareViaShowFileListingSwitch)).check(matches(isChecked()));
-        }
+        //Enable the option for Download/View/Upload permissions
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: Download/View/Upload => Upload only

**Comment by davigonz** [comment]
@jesmrec I requested a few minor changes.

**Comment by jesmrec** [comment]
@davigonz changes applied

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 7]
```diff
@@ -227,35 +231,39 @@ public void test_01_create_public_link_folder_defaults()
...
+        onView(withId(R.id.addPublicLinkButton)).perform(click());
+        onView(withId(R.id.shareViaLinkNameValue)).perform(replaceText(nameShare));
 
-        //Depending the server version, send a name or not.
-        if (capabilities.getVersionMayor() >= VERSION_10) {
-            publicShareCreationAllEnabled(nameShare);
-        } else {
-            publicShareCreationAllEnabled(null);
-        }
+        //Enable the option for Download/View/Upload permissions
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: Download/View/Upload => Download/View ?

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java [hunk 8]
```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
...
     }
 
+
     /**
-     *  TEST CASE: Edit the public folder by enabling "Allow Editing"
-     *  PASSED IF:
-     *             - "Allow editing" and "Show file listing" (oC >= 10) are enabled.
-     *             - "Password" and "Expiration date" are disabled.
+     *  TEST CASE: Share publicly a folder with Upload only permission
+     *  PASSED IF: Link created and visible in share view with Download/View/Upload option
```
Comment (davigonz) [inline androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: Download/View/Upload => Upload only ?

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java (hunk 7 above)
Comment (jesmrec) [reply androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: you are right 👍

File: androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java (hunk 8 above)
Comment (jesmrec) [reply androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java]: 👍

**Pull request merged**